File: Appy/Appy-frontend/src/app/models/client-notifications-settings.ts
import dayjs, { Dayjs } from "dayjs";

export class ClientNotificationsSettingsDTO {
    public instagramAPIAccessToken?: string;
    public appointmentConfirmationMessageTemplate?: string;
    public appointmentReminderMessageTemplate?: string;
    public appointmentReminderTime?: string;
}

export class ClientNotificationsSettings {
    public instagramAPIAccessToken?: string;
    public appointmentConfirmationMessageTemplate?: string;
    public appointmentReminderMessageTemplate?: string;
    public appointmentReminderTime?: Dayjs;

    constructor(dto: ClientNotificationsSettingsDTO = new ClientNotificationsSettingsDTO()) {
        this.instagramAPIAccessToken = dto.instagramAPIAccessToken;
        this.appointmentConfirmationMessageTemplate = dto.appointmentConfirmationMessageTemplate;
        this.appointmentReminderMessageTemplate = dto.appointmentReminderMessageTemplate;
        this.appointmentReminderTime = dto.appointmentReminderTime ? dayjs(dto.appointmentReminderTime) : undefined;
    }

    public getDTO(): ClientNotificationsSettingsDTO {
        let dto = new ClientNotificationsSettingsDTO();
        dto.instagramAPIAccessToken = this.instagramAPIAccessToken;
        dto.appointmentConfirmationMessageTemplate = this.appointmentConfirmationMessageTemplate;
        dto.appointmentReminderMessageTemplate = this.appointmentReminderMessageTemplate;
        dto.appointmentReminderTime = this.appointmentReminderTime?.toISOString();

        return dto;
    }
}

File: Appy/Domain/ClientNotificationsSettings.cs
using Appy.DTOs;
using System.ComponentModel.DataAnnotations;

#pragma warning disable CS8618

namespace Appy.Domain
{
    public class ClientNotificationsSettings
    {
        [Key]
        public int FacilityId { get; set; }

        public string InstagramAPIAccessToken { get; set; }

        public string AppointmentConfirmationMessageTemplate { get; set; }

        /// <summary>
        /// Only time part is used (together with the timezone)
        /// </summary>
        public DateTime? AppointmentReminderTime { get; set; }
        public string AppointmentReminderMessageTemplate { get; set; }

        public ClientNotificationsSettingsDTO GetDTO()
        {
            return new ClientNotificationsSettingsDTO()
            {
                InstagramAPIAccessToken = InstagramAPIAccessToken,
                AppointmentConfirmationMessageTemplate = AppointmentConfirmationMessageTemplate,
                AppointmentReminderTime = AppointmentReminderTime,
                AppointmentReminderMessageTemplate = AppointmentReminderMessageTemplate
            };
        }
    }
}
